KonnectConfigStore: implement the delete op

## Problem Statement

#5187 introduced the `KonnectConfigStore` CRD and wired up its `create` and `update` ops in `crd-from-oas/config.yaml`. The `delete` op was deliberately left out.

If I add `delete: {}` directly, it will fail with the following error.

```
controller/konnect/ops/conditions.go:1: : # github.com/kong/kong-operator/v2/controller/konnect/ops [github.com/kong/kong-operator/v2/controller/konnect/ops.test]
controller/konnect/ops/zz_generated_ops_konnect_configstore.go:78:39: cannot use parentID (variable of type string) as operations.DeleteConfigStoreRequest value in argument to sdk.DeleteConfigStore
controller/konnect/ops/zz_generated_ops_konnect_configstore.go:78:49: cannot use id (variable of type string) as operations.Option value in argument to sdk.DeleteConfigStore (typecheck)
package ops
1 issues:
* typecheck: 1
INFO File cache stats: 1 entries of total size 5.9KiB 
INFO Memory: 1082 samples, avg is 6963.9MB, max is 10429.3MB 
INFO Execution took 1m56.718143542s               
make: *** [generate.lint-fix] Error 1
```

Another thing about the delete API:

`DELETE /v2/control-planes/{controlPlaneId}/config-stores/{configStoreId}` takes a `force` query parameter that **defaults to false**, meaning: refuse to delete the Config Store while it still has secrets linked to it (400). The generator passes `nil` for every optional query parameter, i.e the default.

In the [PRD](https://docs.google.com/document/d/1IDezHAAssWI1dZ55TYwLmhH0oPf8pd6iSLhwVDwgpyE/edit?tab=t.0), some specific boundaries have been defined, and I think we can directly determine how to implement it based on the PRD.



> (_"FR-7 (MUST NOT): The Config Store CRD MUST NOT accept or store secret values; it manages the store container only."_) 

Per FR-7 the CRD never accepts or stores secret values, secrets are written out of band. So "the store has secrets" is the steady state, not an edge case. With force left at its default(`false value`), every delete would return 400; the ops layer only swallows 404, so a 400 becomes a retried failure and we are back at the stuck finalizer



## Proposed Solution

**Generator changes**:

  1. Support wrapped delete requests for single-parent entities , honour `ops.delete.path` the way update does.
  2. Allow generated delete ops to pass non-nil values for delete query parameters, so force can be set at all. (just a field)

Then add the delete stanza for `KonnectConfigStore` and regenerate.

  **Pass**` force=true`. Rationale:

  - Leaving it at the default makes deletion fail in the normal case and wedges the finalizer, violating FR-12(FR-12 (MUST): All resources in the chain MUST delete cleanly regardless of deletion order, without stuck finalizers requiring manual intervention.).
  - FR-7 constrains the CRD's API surface — it must not accept or store secret values. FR-5 explicitly puts creating and deleting the store
  container in scope. Konnect cascading to the container's contents on delete is container lifecycle, and matches ordinary Kubernetes
  semantics for deleting an owner.
  - Per NFR-4 this stays purely additive: no new API field now. If an opt-out is ever needed, adding an optional field later is backward
  compatible.

  **Do not block deletion on in-use references**. The real hazard is a user deleting only the Config Store CR while a KongVault /
  KongCertificate still references it. The instinctive guard — refuse to delete while referenced — would violate FR-12. The PRD puts this on
  the reference layer instead: FR-10 requires a KongVault pointing at an unavailable Config Store to surface that in its status and
  reconcile automatically once it is ready again.

  Document that deleting a KonnectConfigStore CR deletes every secret in that store in Konnect (NFR-2). This is the one genuinely surprising
  consequence and it must be explicit in the docs.


## Additional Information

...

## Acceptance Criteria

<!--
Updates the acceptance criteria below!
If this feature / enhancement does not require changes in docs / tests / helm chart,
cross out the corresponding items below, ideally commenting why they are not needed.
-->

- [ ] Documentation is updated with this new feature / enhancement
- [ ] Tests are added for this feature / enhancement
- [ ] Helm chart is updated to work with this feature / enhancement
- [ ] As a user, I can ...

--- a/controller/konnect/ops/zz_generated_ops_delete.go
+++ b/controller/konnect/ops/zz_generated_ops_delete.go
@@ -65,6 +65,8 @@ func DeleteGeneratedOps[
 		return deleteEventGatewayVirtualClusterProducePolicy(ctx, sdk.GetEventGatewayVirtualClusterProducePoliciesSDK(), ent)
 	case *konnectv1alpha1.KonnectAIGateway:
 		return deleteKonnectAIGateway(ctx, sdk.GetAIGatewaysSDK(), ent)
+	case *konnectv1alpha1.KonnectConfigStore:
+		return deleteKonnectConfigStore(ctx, sdk.GetConfigStoresSDK(), ent)
 	case *konnectv1alpha1.KonnectEventGateway:
 		return deleteKonnectEventGateway(ctx, sdk.GetEventGatewaysSDK(), ent)
 	case *konnectv1alpha1.Portal:

--- a/controller/konnect/ops/zz_generated_ops_konnect_configstore.go
+++ b/controller/konnect/ops/zz_generated_ops_konnect_configstore.go
@@ -64,6 +64,28 @@ func updateKonnectConfigStore(
 	return nil
 }
 
+func deleteKonnectConfigStore(
+	ctx context.Context,
+	sdk sdkkonnectgo.ConfigStoresSDK,
+	obj *konnectv1alpha1.KonnectConfigStore,
+) error {
+	parentID := obj.GetControlPlaneID()
+	if parentID == "" {
+		return CantPerformOperationWithoutParentIDError{Entity: obj, Parent: "KonnectGatewayControlPlane", Op: DeleteOp}
+	}
+	id := obj.GetKonnectStatus().GetKonnectID()
+
+	_, err := sdk.DeleteConfigStore(ctx, sdkkonnectops.DeleteConfigStoreRequest{
+		ControlPlaneID: parentID,
+		ConfigStoreID:  id,
+		Force:          sdkkonnectops.ForceTrue.ToPointer(),
+	})
+	if errWrap := wrapErrIfKonnectOpFailed(err, DeleteOp, obj); errWrap != nil {
+		return handleDeleteError(ctx, errWrap, obj)
+	}
+	return nil
+}
+
 func getKonnectConfigStoreForUID(
 	ctx context.Context,
 	sdk sdkkonnectgo.ConfigStoresSDK,

--- a/controller/konnect/ops/zz_generated_ops_konnect_configstore_test.go
+++ b/controller/konnect/ops/zz_generated_ops_konnect_configstore_test.go
@@ -140,3 +140,55 @@ func TestUpdateKonnectConfigStore_PropagatesSDKError(t *testing.T) {
 	err = updateKonnectConfigStore(ctx, sdk, obj)
 	require.ErrorContains(t, err, sdkErr.Error())
 }
+
+func TestDeleteKonnectConfigStore_UsesGeneratedSDKOps(t *testing.T) {
+	t.Parallel()
+
+	ctx := t.Context()
+	sdk := mocks.NewMockConfigStoresSDK(t)
+	obj := testGeneratedKonnectConfigStoreForSDKOps()
+	parentID := "parentID-1"
+	obj.SetControlPlaneID(parentID)
+	obj.SetKonnectID("konnect_configstore-id")
+
+	sdk.EXPECT().
+		DeleteConfigStore(
+			mock.Anything,
+			sdkkonnectops.DeleteConfigStoreRequest{
+				ControlPlaneID: parentID,
+				ConfigStoreID:  obj.GetKonnectStatus().GetKonnectID(),
+				Force:          sdkkonnectops.ForceTrue.ToPointer(),
+			},
+		).
+		Return(&sdkkonnectops.DeleteConfigStoreResponse{}, nil).
+		Once()
+
+	require.NoError(t, deleteKonnectConfigStore(ctx, sdk, obj))
+}
+
+func TestDeleteKonnectConfigStore_PropagatesSDKError(t *testing.T) {
+	t.Parallel()
+
+	ctx := t.Context()
+	sdk := mocks.NewMockConfigStoresSDK(t)
+	obj := testGeneratedKonnectConfigStoreForSDKOps()
+	parentID := "parentID-1"
+	obj.SetControlPlaneID(parentID)
+	obj.SetKonnectID("konnect_configstore-id")
+	sdkErr := errors.New("sdk error")
+
+	sdk.EXPECT().
+		DeleteConfigStore(
+			mock.Anything,
+			sdkkonnectops.DeleteConfigStoreRequest{
+				ControlPlaneID: parentID,
+				ConfigStoreID:  obj.GetKonnectStatus().GetKonnectID(),
+				Force:          sdkkonnectops.ForceTrue.ToPointer(),
+			},
+		).
+		Return(nil, sdkErr).
+		Once()
+
+	err := deleteKonnectConfigStore(ctx, sdk, obj)
+	require.ErrorContains(t, err, sdkErr.Error())
+}

--- a/controller/konnect/reconciler_apiauthref_get.go
+++ b/controller/konnect/reconciler_apiauthref_get.go
@@ -14,6 +14,7 @@ import (
 	commonv1alpha1 "github.com/kong/kong-operator/v2/api/common/v1alpha1"
 	configurationv1alpha1 "github.com/kong/kong-operator/v2/api/configuration/v1alpha1"
 	konnectv1alpha1 "github.com/kong/kong-operator/v2/api/konnect/v1alpha1"
+	konnectv1alpha2 "github.com/kong/kong-operator/v2/api/konnect/v1alpha2"
 	"github.com/kong/kong-operator/v2/controller/konnect/constraints"
 )
 
@@ -35,6 +36,15 @@ type eventGatewayVirtualClusterRefAccessor interface {
 type eventGatewayListenerRefAccessor interface {
 	objectWithParentRef
 	GetEventGatewayListenerRef() commonv1alpha1.ObjectRef
+}
+
+// konnectGatewayControlPlaneRefAccessor is implemented by generated entities whose
+// parent is a KonnectGatewayControlPlane referenced through a generated ObjectRef
+// (e.g. KonnectConfigStore). Hand-written entities reference their Control Plane
+// through commonv1alpha1.ControlPlaneRef instead, so they do not match here.
+type konnectGatewayControlPlaneRefAccessor interface {
+	objectWithParentRef
+	GetControlPlaneRef() commonv1alpha1.ObjectRef
 }
 
 type portalRefAccessor interface {
@@ -95,6 +105,9 @@ func getAPIAuthRef[
 	// Other Konnect entities.
 	if obj, ok := any(ent).(portalRefAccessor); ok {
 		return getAPIAuthConfigurationRefFromParent[konnectv1alpha1.Portal](ctx, cl, obj, obj.GetParentRef())
+	}
+	if obj, ok := any(ent).(konnectGatewayControlPlaneRefAccessor); ok {
+		return getAPIAuthConfigurationRefFromParent[konnectv1alpha2.KonnectGatewayControlPlane](ctx, cl, obj, obj.GetParentRef())
 	}
 
 	return types.NamespacedName{},

--- a/crd-from-oas/config.yaml
+++ b/crd-from-oas/config.yaml
@@ -841,6 +841,19 @@ apiGroupVersions:
             path: github.com/Kong/sdk-konnect-go/models/components.CreateConfigStore
           update:
             path: github.com/Kong/sdk-konnect-go/models/operations.UpdateConfigStoreRequest
+          delete:
+            path: github.com/Kong/sdk-konnect-go/models/operations.DeleteConfigStoreRequest
+            requestFields:
+              # Delete the config store even when it still holds secret entries.
+              # The operator never manages those entries, so they are always
+              # written out of band and are present in the common case. With
+              # force unset Konnect rejects the delete with a 400, which would
+              # leave the CR's cleanup finalizer in place forever.
+              # NOTE: this means deleting a KonnectConfigStore deletes every
+              # secret in that store on the Konnect side. This must be called
+              # out in the user-facing docs.
+              - name: Force
+                value: sdkkonnectops.ForceTrue.ToPointer()
         reconciler:
           parentEntityGVK:
             kind: KonnectGatewayControlPlane

--- a/crd-from-oas/pkg/config/config.go
+++ b/crd-from-oas/pkg/config/config.go
@@ -376,10 +376,33 @@ type OpConfig struct {
 	// Path is the fully qualified Go type path in the form "importpath.TypeName",
 	// e.g. "github.com/Kong/sdk-konnect-go/models/components.CreatePortal".
 	// Required for create/update ops.
+	//
+	// For delete ops Path is optional and must reference a type in the SDK's
+	// models/operations package. Setting it opts the entity into the wrapped
+	// request-struct call shape, which entities with two or more parents get
+	// automatically. Single-parent entities need it whenever the SDK's delete
+	// method takes an operations.DeleteXxxRequest rather than positional
+	// arguments.
 	Path string `yaml:"path,omitempty"`
 	// AsPUT makes generated delete ops call the configured update/PUT SDK method
 	// with an empty request body instead of requiring an OpenAPI DELETE operation.
 	AsPUT bool `yaml:"asPUT,omitempty"`
+	// RequestFields assigns constant values to additional fields on a wrapped
+	// request struct, e.g. an optional query parameter that the generator would
+	// otherwise leave unset. Only supported for delete ops using the wrapped
+	// call shape.
+	RequestFields []RequestFieldConfig `yaml:"requestFields,omitempty"`
+}
+
+// RequestFieldConfig assigns a constant value to a single field of an SDK
+// request struct.
+type RequestFieldConfig struct {
+	// Name is the Go field name on the SDK request struct, e.g. "Force".
+	Name string `yaml:"name"`
+	// Value is the Go expression assigned to the field. It is emitted verbatim,
+	// so it must be valid in the generated ops package, e.g.
+	// "sdkkonnectops.ForceTrue.ToPointer()".
+	Value string `yaml:"value"`
 }
 
 // OpSDKConfig identifies the SDK interface and factory field name used by
@@ -941,10 +964,23 @@ func (tc *TypeConfig) validate() error {
 		}
 		seenSecretPaths[sr.Path] = true
 	}
-	if deleteOp, ok := tc.Ops["delete"]; ok && deleteOp != nil && deleteOp.AsPUT {
-		updateOp, ok := tc.Ops["update"]
-		if !ok || updateOp == nil || updateOp.Path == "" {
-			return fmt.Errorf("ops.delete.asPUT requires ops.update.path")
+	if deleteOp, ok := tc.Ops["delete"]; ok && deleteOp != nil {
+		if deleteOp.AsPUT {
+			updateOp, ok := tc.Ops["update"]
+			if !ok || updateOp == nil || updateOp.Path == "" {
+				return fmt.Errorf("ops.delete.asPUT requires ops.update.path")
+			}
+			if deleteOp.Path != "" {
+				return fmt.Errorf("ops.delete.asPUT and ops.delete.path are mutually exclusive")
+			}
+		}
+		for i, f := range deleteOp.RequestFields {
+			if f.Name == "" {
+				return fmt.Errorf("ops.delete.requestFields[%d].name is required", i)
+			}
+			if f.Value == "" {
+				return fmt.Errorf("ops.delete.requestFields[%d].value is required", i)
+			}
 		}
 	}
 	if tc.OpsSkipGetForUID && tc.OpsGetForUID != nil {

--- a/crd-from-oas/pkg/config/config_test.go
+++ b/crd-from-oas/pkg/config/config_test.go
@@ -164,6 +164,86 @@ apiGroupVersions:
 		assert.Contains(t, err.Error(), "ops.delete.asPUT requires ops.update.path")
 	})
 
+	t.Run("valid config with delete path and requestFields", func(t *testing.T) {
+		content := `
+apiGroupVersions:
+  konnect.konghq.com/v1alpha1:
+    types:
+      - path: /v2/control-planes/{controlPlaneId}/config-stores
+        ops:
+          create:
+            path: github.com/Kong/sdk-konnect-go/models/components.CreateConfigStore
+          delete:
+            path: github.com/Kong/sdk-konnect-go/models/operations.DeleteConfigStoreRequest
+            requestFields:
+              - name: Force
+                value: sdkkonnectops.ForceTrue.ToPointer()
+`
+		path := filepath.Join(t.TempDir(), "config.yaml")
+		require.NoError(t, os.WriteFile(path, []byte(content), 0o600))
+
+		cfg, err := LoadProjectConfig(path)
+		require.NoError(t, err)
+
+		konnect := cfg.APIGroupVersions["konnect.konghq.com/v1alpha1"]
+		require.NotNil(t, konnect)
+		require.Len(t, konnect.Types, 1)
+		deleteOp := konnect.Types[0].Ops["delete"]
+		require.NotNil(t, deleteOp)
+		assert.Equal(t,
+			"github.com/Kong/sdk-konnect-go/models/operations.DeleteConfigStoreRequest",
+			deleteOp.Path,
+		)
+		require.Len(t, deleteOp.RequestFields, 1)
+		assert.Equal(t, "Force", deleteOp.RequestFields[0].Name)
+		assert.Equal(t, "sdkkonnectops.ForceTrue.ToPointer()", deleteOp.RequestFields[0].Value)
+	})
+
+	t.Run("invalid config with delete asPUT and path", func(t *testing.T) {
+		content := `
+apiGroupVersions:
+  konnect.konghq.com/v1alpha1:
+    types:
+      - path: /v3/portals/{portalId}/customization
+        ops:
+          create:
+            path: github.com/Kong/sdk-konnect-go/models/components.PortalCustomization
+          update:
+            path: github.com/Kong/sdk-konnect-go/models/components.PortalCustomization
+          delete:
+            asPUT: true
+            path: github.com/Kong/sdk-konnect-go/models/operations.DeletePortalCustomizationRequest
+`
+		path := filepath.Join(t.TempDir(), "config.yaml")
+		require.NoError(t, os.WriteFile(path, []byte(content), 0o600))
+
+		_, err := LoadProjectConfig(path)
+		require.Error(t, err)
+		assert.Contains(t, err.Error(), "ops.delete.asPUT and ops.delete.path are mutually exclusive")
+	})
+
+	t.Run("invalid config with delete requestFields missing value", func(t *testing.T) {
+		content := `
+apiGroupVersions:
+  konnect.konghq.com/v1alpha1:
+    types:
+      - path: /v2/control-planes/{controlPlaneId}/config-stores
+        ops:
+          create:
+            path: github.com/Kong/sdk-konnect-go/models/components.CreateConfigStore
+          delete:
+            path: github.com/Kong/sdk-konnect-go/models/operations.DeleteConfigStoreRequest
+            requestFields:
+              - name: Force
+`
+		path := filepath.Join(t.TempDir(), "config.yaml")
+		require.NoError(t, os.WriteFile(path, []byte(content), 0o600))
+
+		_, err := LoadProjectConfig(path)
+		require.Error(t, err)
+		assert.Contains(t, err.Error(), "ops.delete.requestFields[0].value is required")
+	})
+
 	t.Run("valid config with ops uid tag filter", func(t *testing.T) {
 		content := `
 apiGroupVersions:

--- a/crd-from-oas/pkg/generator/generator_test.go
+++ b/crd-from-oas/pkg/generator/generator_test.go
@@ -5770,6 +5770,105 @@ func TestGenerateOpsDelete_NonRootEntity(t *testing.T) {
 	assert.NotContains(t, file.Content, "sdkkonnectops.Delete")
 }
 
+// newConfigStoreDeleteFixture returns the generator, schema and ops config for a
+// single-parent entity whose SDK delete method takes a wrapped request struct.
+func newConfigStoreDeleteFixture(deleteOp *config.OpConfig) (*Generator, *parser.Schema, *config.EntityOpsConfig) {
+	g := NewGenerator(Config{
+		APIGroup:             "konnect.konghq.com",
+		APIVersion:           "v1alpha1",
+		APIGroupPackagePath:  "github.com/kong/kong-operator/v2/api/konnect/v1alpha1",
+		APIGroupPackageAlias: "konnectv1alpha1",
+		ReconcilerConfig: map[string]*config.ReconcilerConfig{
+			"KonnectConfigStore": {
+				IsRoot: new(false),
+				ParentEntityGVK: &config.EntityGVKConfig{
+					Kind:    "KonnectGatewayControlPlane",
+					Group:   "konnect.konghq.com",
+					Version: "v1alpha2",
+				},
+			},
+		},
+	})
+
+	schema := &parser.Schema{
+		OperationID:        "create-config-store",
+		Tags:               []string{"Config Stores"},
+		SuccessResponseRef: "ConfigStoreResponse",
+		Dependencies: []*parser.Dependency{
+			{ParamName: "controlPlaneId", EntityName: "ControlPlane"},
+		},
+		// DELETE /v2/control-planes/{controlPlaneId}/config-stores/{configStoreId}
+		// — 2 path params, 1 optional query param (force).
+		DeleteOperationID:     "delete-config-store",
+		DeleteTags:            []string{"Config Stores"},
+		DeletePathParams:      []string{"controlPlaneId", "configStoreId"},
+		DeleteQueryParamCount: 1,
+	}
+	opsConfig := &config.EntityOpsConfig{
+		SDK: &config.OpSDKConfig{
+			Interface: "github.com/Kong/sdk-konnect-go.ConfigStoresSDK",
+			FieldName: "ConfigStores",
+		},
+		Ops: map[string]*config.OpConfig{
+			"create": {Path: "github.com/Kong/sdk-konnect-go/models/components.CreateConfigStore"},
+			"delete": deleteOp,
+		},
+	}
+	return g, schema, opsConfig
+}
+
+func TestGenerateOpsDelete_SingleParentWrappedRequestFromConfiguredPath(t *testing.T) {
+	g, schema, opsConfig := newConfigStoreDeleteFixture(&config.OpConfig{
+		Path: "github.com/Kong/sdk-konnect-go/models/operations.DeleteConfigStoreRequest",
+		RequestFields: []config.RequestFieldConfig{
+			{Name: "Force", Value: "sdkkonnectops.ForceTrue.ToPointer()"},
+		},
+	})
+
+	file, info, err := g.generateOpsDelete("KonnectConfigStore", schema, opsConfig)
+	require.NoError(t, err)
+	require.NotNil(t, file)
+	require.NotNil(t, info)
+
+	assert.Equal(t, "GetConfigStoresSDK", info.SDKGetter)
+
+	// A single parent still gets the wrapped call shape because ops.delete.path
+	// names a type in the SDK operations package. The configured type name is
+	// used verbatim rather than the "<Method>Request" convention.
+	assert.Contains(t, file.Content, "sdk.DeleteConfigStore(ctx, sdkkonnectops.DeleteConfigStoreRequest{")
+	assert.Contains(t, file.Content, "ControlPlaneID: parentID,")
+	assert.Contains(t, file.Content, "ConfigStoreID: id,")
+
+	// Configured request fields are emitted verbatim, so the optional force query
+	// param is set instead of being left nil.
+	assert.Contains(t, file.Content, "Force: sdkkonnectops.ForceTrue.ToPointer(),")
+
+	// The positional call shape (and its nil query param) is not emitted.
+	assert.NotContains(t, file.Content, "sdk.DeleteConfigStore(ctx, parentID, id")
+}
+
+func TestGenerateOpsDelete_ConfiguredPathMustBeOperationsType(t *testing.T) {
+	g, schema, opsConfig := newConfigStoreDeleteFixture(&config.OpConfig{
+		Path: "github.com/Kong/sdk-konnect-go/models/components.DeleteConfigStore",
+	})
+
+	_, _, err := g.generateOpsDelete("KonnectConfigStore", schema, opsConfig)
+	require.Error(t, err)
+	assert.Contains(t, err.Error(), "ops.delete.path must reference a type in the SDK operations package")
+}
+
+func TestGenerateOpsDelete_RequestFieldsRequireWrappedCallShape(t *testing.T) {
+	g, schema, opsConfig := newConfigStoreDeleteFixture(&config.OpConfig{
+		RequestFields: []config.RequestFieldConfig{
+			{Name: "Force", Value: "sdkkonnectops.ForceTrue.ToPointer()"},
+		},
+	})
+
+	_, _, err := g.generateOpsDelete("KonnectConfigStore", schema, opsConfig)
+	require.Error(t, err)
+	assert.Contains(t, err.Error(), "ops.delete.requestFields requires the wrapped call shape")
+}
+
 func TestGenerateOpsDelete_NonRootEntityWithParentTypeOverride(t *testing.T) {
 	g := NewGenerator(Config{
 		APIGroupPackagePath:  "github.com/kong/kong-operator/v2/api/konnect/v1alpha1",

--- a/crd-from-oas/pkg/generator/ops_delete.go
+++ b/crd-from-oas/pkg/generator/ops_delete.go
@@ -2,6 +2,7 @@ package generator
 
 import (
 	"fmt"
+	"strings"
 
 	"github.com/kong/kong-operator/v2/crd-from-oas/pkg/config"
 	"github.com/kong/kong-operator/v2/crd-from-oas/pkg/parser"
@@ -34,6 +35,10 @@ type opsDeleteFuncData struct {
 	// DeleteEntityIDField is the SDK request field name for the entity's own ID,
 	// e.g. "PolicyID". Used only when DeleteFullyWrapped is true.
 	DeleteEntityIDField string
+	// DeleteRequestFields holds constant assignments applied to the wrapped delete
+	// request struct, e.g. the `force` query parameter on config store deletion.
+	// Used only when DeleteFullyWrapped is true.
+	DeleteRequestFields []config.RequestFieldConfig
 	// DeleteNilArgs holds one entry per optional query parameter on the DELETE
 	// operation. The SDK codegen promotes query params to positional args before
 	// the variadic opts; we pass nil for each since they are all optional.
@@ -128,8 +133,29 @@ func (g *Generator) generateOpsDeleteFuncBody(
 		return nil, err
 	}
 
-	// Multi-parent entities use a fully-wrapped delete request struct.
+	// Multi-parent entities always use a fully-wrapped delete request struct.
+	// Single-parent entities whose SDK delete method takes one too opt in by
+	// configuring ops.delete.path, mirroring how the update op resolves its call
+	// shape from the configured request type.
 	deleteFullyWrapped := len(parents) >= 2
+	var configuredWrappedType string
+	if deleteOp.Path != "" {
+		importPath, typeName, err := ParseSDKTypePath(deleteOp.Path)
+		if err != nil {
+			return nil, fmt.Errorf("entity %q: %w", entityName, err)
+		}
+		if !strings.HasSuffix(importPath, "/operations") {
+			return nil, fmt.Errorf(
+				"entity %q: ops.delete.path must reference a type in the SDK operations package, got %q",
+				entityName, importPath,
+			)
+		}
+		deleteFullyWrapped = true
+		configuredWrappedType = typeName
+	}
+	if len(deleteOp.RequestFields) > 0 && !deleteFullyWrapped {
+		return nil, fmt.Errorf("entity %q: ops.delete.requestFields requires the wrapped call shape (set ops.delete.path)", entityName)
+	}
 	// deleteOmitsEntityID is true for parent-scoped singletons: the DELETE path
 	// contains only parent path params (no entity-specific ID). The SDK method
 	// takes the parent ID positionally; no entity ID is emitted.
@@ -140,6 +166,9 @@ func (g *Generator) generateOpsDeleteFuncBody(
 		// The wrapped request type name follows the SDK convention:
 		// "<PascalCaseDeleteMethod>Request", e.g. "DeleteEventGatewayListenerPolicyRequest".
 		deleteWrappedType = sdkMethod + "Request"
+		if configuredWrappedType != "" {
+			deleteWrappedType = configuredWrappedType
+		}
 		// Entity ID field is derived from the last delete path param.
 		if len(schema.DeletePathParams) > 0 {
 			deleteEntityIDField = pathParamToFieldName(schema.DeletePathParams[len(schema.DeletePathParams)-1])
@@ -157,6 +186,7 @@ func (g *Generator) generateOpsDeleteFuncBody(
 		DeleteFullyWrapped:  deleteFullyWrapped,
 		DeleteWrappedType:   deleteWrappedType,
 		DeleteEntityIDField: deleteEntityIDField,
+		DeleteRequestFields: deleteOp.RequestFields,
 		DeleteNilArgs:       nilArgs,
 		DeleteOmitsEntityID: deleteOmitsEntityID,
 		SupportsMirror:      g.entitySupportsMirror(entityName),

--- a/crd-from-oas/pkg/generator/templates.go
+++ b/crd-from-oas/pkg/generator/templates.go
@@ -2221,6 +2221,9 @@ func TestDelete{{.Entity}}_UsesGeneratedSDKOps(t *testing.T) {
 				{{.SDKFieldName}}: {{.VarName}},
 {{- end}}
 				{{.Delete.DeleteEntityIDField}}: obj.GetKonnectStatus().GetKonnectID(),
+{{- range .Delete.DeleteRequestFields}}
+				{{.Name}}: {{.Value}},
+{{- end}}
 			},
 {{- else if .Delete.DeleteOmitsEntityID}}
 			{{(index .Delete.Parents 0).VarName}}{{range .Delete.DeleteNilArgs}}, mock.Anything{{end}},
@@ -2292,6 +2295,9 @@ func TestDelete{{.Entity}}_PropagatesSDKError(t *testing.T) {
 				{{.SDKFieldName}}: {{.VarName}},
 {{- end}}
 				{{.Delete.DeleteEntityIDField}}: obj.GetKonnectStatus().GetKonnectID(),
+{{- range .Delete.DeleteRequestFields}}
+				{{.Name}}: {{.Value}},
+{{- end}}
 			},
 {{- else if .Delete.DeleteOmitsEntityID}}
 			{{(index .Delete.Parents 0).VarName}}{{range .Delete.DeleteNilArgs}}, mock.Anything{{end}},
@@ -2740,6 +2746,9 @@ func delete{{.Entity}}(
 		{{.SDKFieldName}}: {{.VarName}},
 		{{- end}}
 		{{.DeleteEntityIDField}}: id,
+		{{- range .DeleteRequestFields}}
+		{{.Name}}: {{.Value}},
+		{{- end}}
 	})
 {{- else if and (not .DeleteAsUpdate) .DeleteOmitsEntityID}}
 

--- a/test/envtest/konnect-other/konnect_entities_konnectconfigstore_test.go
+++ b/test/envtest/konnect-other/konnect_entities_konnectconfigstore_test.go
@@ -1,0 +1,155 @@
+package konnectother
+
+import (
+	"testing"
+
+	sdkkonnectcomp "github.com/Kong/sdk-konnect-go/models/components"
+	sdkkonnectops "github.com/Kong/sdk-konnect-go/models/operations"
+	sdkkonnecterrs "github.com/Kong/sdk-konnect-go/models/sdkerrors"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	apiwatch "k8s.io/apimachinery/pkg/watch"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	konnectv1alpha1 "github.com/kong/kong-operator/v2/api/konnect/v1alpha1"
+	"github.com/kong/kong-operator/v2/controller/konnect"
+	"github.com/kong/kong-operator/v2/modules/manager/logging"
+	"github.com/kong/kong-operator/v2/modules/manager/scheme"
+	"github.com/kong/kong-operator/v2/test/envtest"
+	"github.com/kong/kong-operator/v2/test/envtest/consts"
+	"github.com/kong/kong-operator/v2/test/helpers/deploy"
+	"github.com/kong/kong-operator/v2/test/helpers/eventually"
+	"github.com/kong/kong-operator/v2/test/mocks/metricsmocks"
+	"github.com/kong/kong-operator/v2/test/mocks/sdkmocks"
+)
+
+func TestKonnectConfigStore(t *testing.T) {
+	t.Parallel()
+	ctx, cancel := envtest.Context(t, t.Context())
+	defer cancel()
+	cfg, ns := envtest.Setup(t, ctx, scheme.Get(), envtest.WithInstallGatewayCRDs(true))
+
+	t.Log("Setting up the manager with reconcilers")
+	mgr, logs := envtest.NewManager(t, ctx, cfg, scheme.Get())
+	factory := sdkmocks.NewMockSDKFactory(t)
+	sdk := factory.SDK
+	envtest.StartReconcilers(ctx, t, mgr, logs,
+		konnect.NewKonnectEntityReconciler(factory, logging.DevelopmentMode, mgr.GetClient(),
+			konnect.WithKonnectEntitySyncPeriod[konnectv1alpha1.KonnectConfigStore](consts.KonnectInfiniteSyncTime),
+			konnect.WithMetricRecorder[konnectv1alpha1.KonnectConfigStore](&metricsmocks.MockRecorder{}),
+		),
+	)
+
+	t.Log("Setting up clients")
+	cl, err := client.NewWithWatch(mgr.GetConfig(), client.Options{
+		Scheme: scheme.Get(),
+	})
+	require.NoError(t, err)
+	clientNamespaced := client.NewNamespacedClient(mgr.GetClient(), ns.Name)
+
+	t.Log("Creating KonnectAPIAuthConfiguration and parent KonnectGatewayControlPlane")
+	apiAuth := deploy.KonnectAPIAuthConfigurationWithProgrammed(t, ctx, clientNamespaced)
+	cp := deploy.KonnectGatewayControlPlaneWithID(t, ctx, clientNamespaced, apiAuth)
+
+	// Deleting a config store that still holds secret entries is the common case:
+	// the operator never manages those entries, so they are always written out of
+	// band. The delete op must therefore force the deletion, otherwise Konnect
+	// rejects it and the cleanup finalizer is never released.
+	t.Run("should create and force-delete KonnectConfigStore successfully", func(t *testing.T) {
+		const configStoreID = "config-store-12345"
+
+		w := envtest.SetupWatch[konnectv1alpha1.KonnectConfigStoreList](t, ctx, cl, client.InNamespace(ns.Name))
+
+		t.Log("Setting up SDK expectations on KonnectConfigStore creation")
+		sdk.ConfigStoresSDK.EXPECT().
+			CreateConfigStore(mock.Anything, cp.GetKonnectID(), mock.Anything).
+			Return(&sdkkonnectops.CreateConfigStoreResponse{
+				ConfigStore: &sdkkonnectcomp.ConfigStore{
+					ID: new(configStoreID),
+				},
+			}, nil).
+			Once()
+
+		t.Log("Creating KonnectConfigStore")
+		configStore := deploy.KonnectConfigStore(t, ctx, clientNamespaced, cp)
+
+		t.Log("Waiting for KonnectConfigStore to be programmed")
+		envtest.WatchFor(t, ctx, w, apiwatch.Modified,
+			envtest.AssertsAnd(
+				envtest.ObjectMatchesName(configStore),
+				envtest.ObjectMatchesKonnectID[*konnectv1alpha1.KonnectConfigStore](configStoreID),
+				envtest.ObjectHasConditionProgrammedSetToTrue[*konnectv1alpha1.KonnectConfigStore](),
+				func(cs *konnectv1alpha1.KonnectConfigStore) bool {
+					return cs.GetControlPlaneID() == cp.GetKonnectID() &&
+						controllerutil.ContainsFinalizer(cs, konnect.KonnectCleanupFinalizer)
+				},
+			),
+			"KonnectConfigStore didn't get Programmed status condition, Konnect ID, Control Plane ID, or cleanup finalizer",
+		)
+
+		envtest.EventuallyAssertSDKExpectations(t, sdk.ConfigStoresSDK, consts.WaitTime, consts.TickTime)
+
+		t.Log("Setting up SDK expectations on KonnectConfigStore deletion")
+		sdk.ConfigStoresSDK.EXPECT().
+			DeleteConfigStore(mock.Anything, mock.MatchedBy(func(req sdkkonnectops.DeleteConfigStoreRequest) bool {
+				return req.ControlPlaneID == cp.GetKonnectID() &&
+					req.ConfigStoreID == configStoreID &&
+					req.Force != nil && *req.Force == sdkkonnectops.ForceTrue
+			})).
+			Return(&sdkkonnectops.DeleteConfigStoreResponse{}, nil).
+			Once()
+
+		t.Log("Deleting KonnectConfigStore")
+		require.NoError(t, clientNamespaced.Delete(ctx, configStore))
+		eventually.WaitForObjectToNotExist(t, ctx, clientNamespaced, configStore, consts.WaitTime, consts.TickTime)
+		envtest.EventuallyAssertSDKExpectations(t, sdk.ConfigStoresSDK, consts.WaitTime, consts.TickTime)
+	})
+
+	// The config store can already be gone on the Konnect side, e.g. when the
+	// parent Control Plane was deleted first. The cleanup finalizer must still be
+	// released so the CR does not linger in Terminating.
+	t.Run("should release the cleanup finalizer when the config store is already gone from Konnect", func(t *testing.T) {
+		const configStoreID = "config-store-already-gone"
+
+		w := envtest.SetupWatch[konnectv1alpha1.KonnectConfigStoreList](t, ctx, cl, client.InNamespace(ns.Name))
+
+		sdk.ConfigStoresSDK.EXPECT().
+			CreateConfigStore(mock.Anything, cp.GetKonnectID(), mock.Anything).
+			Return(&sdkkonnectops.CreateConfigStoreResponse{
+				ConfigStore: &sdkkonnectcomp.ConfigStore{
+					ID: new(configStoreID),
+				},
+			}, nil).
+			Once()
+
+		configStore := deploy.KonnectConfigStore(t, ctx, clientNamespaced, cp)
+
+		envtest.WatchFor(t, ctx, w, apiwatch.Modified,
+			envtest.AssertsAnd(
+				envtest.ObjectMatchesName(configStore),
+				envtest.ObjectMatchesKonnectID[*konnectv1alpha1.KonnectConfigStore](configStoreID),
+				envtest.ObjectHasConditionProgrammedSetToTrue[*konnectv1alpha1.KonnectConfigStore](),
+			),
+			"KonnectConfigStore didn't get Programmed status condition or Konnect ID",
+		)
+
+		envtest.EventuallyAssertSDKExpectations(t, sdk.ConfigStoresSDK, consts.WaitTime, consts.TickTime)
+
+		t.Log("Setting up SDK expectations on KonnectConfigStore deletion returning 404")
+		sdk.ConfigStoresSDK.EXPECT().
+			DeleteConfigStore(mock.Anything, mock.MatchedBy(func(req sdkkonnectops.DeleteConfigStoreRequest) bool {
+				return req.ConfigStoreID == configStoreID
+			})).
+			Return(nil, &sdkkonnecterrs.SDKError{
+				StatusCode: 404,
+				Message:    "not found",
+			}).
+			Once()
+
+		t.Log("Deleting KonnectConfigStore")
+		require.NoError(t, clientNamespaced.Delete(ctx, configStore))
+		eventually.WaitForObjectToNotExist(t, ctx, clientNamespaced, configStore, consts.WaitTime, consts.TickTime)
+		envtest.EventuallyAssertSDKExpectations(t, sdk.ConfigStoresSDK, consts.WaitTime, consts.TickTime)
+	})
+}

--- a/test/helpers/deploy/deploy_resources.go
+++ b/test/helpers/deploy/deploy_resources.go
@@ -661,6 +661,43 @@ func EventGatewayBackendCluster(
 	return &obj
 }
 
+// KonnectConfigStore deploys a KonnectConfigStore resource scoped to the given
+// KonnectGatewayControlPlane and returns it.
+func KonnectConfigStore(
+	t *testing.T,
+	ctx context.Context,
+	cl client.Client,
+	cp *konnectv1alpha2.KonnectGatewayControlPlane,
+	opts ...ObjOption,
+) *konnectv1alpha1.KonnectConfigStore {
+	t.Helper()
+	name := "config-store-" + randomSuffix()
+	obj := konnectv1alpha1.KonnectConfigStore{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: konnectv1alpha1.KonnectConfigStoreSpec{
+			ControlPlaneRef: commonv1alpha1.ObjectRef{
+				Type: commonv1alpha1.ObjectRefTypeNamespacedRef,
+				NamespacedRef: &commonv1alpha1.NamespacedRef{
+					Name: cp.Name,
+				},
+			},
+			APISpec: konnectv1alpha1.KonnectConfigStoreAPISpec{
+				Name: name,
+			},
+		},
+	}
+
+	for _, opt := range opts {
+		opt(&obj)
+	}
+
+	require.NoError(t, cl.Create(ctx, &obj))
+	logObjectCreate(t, &obj)
+	return &obj
+}
+
 // KonnectCloudGatewayNetwork deploys a KonnectCloudGatewayNetwork resource and returns it.
 func KonnectCloudGatewayNetwork(
 	t *testing.T,

